src/features/game/quiz/GameQuizManager.ts

```diff
@@ -0,0 +1,214 @@
+import ImageAssets from '../assets/ImageAssets';
+import SoundAssets from '../assets/SoundAssets';
+import { Constants, screenSize } from '../commons/CommonConstants';
+import { ItemId } from '../commons/CommonTypes';
+import { createDialogueBox, createTypewriter } from '../dialogue/GameDialogueHelper';
+import { DialogueObject } from '../dialogue/GameDialogueTypes';
+import { fadeAndDestroy } from '../effects/FadeEffect';
+import { rightSideEntryTweenProps, rightSideExitTweenProps } from '../effects/FlyEffect';
+import { promptWithChoices } from '../effects/Prompt';
+import { Layer } from '../layer/GameLayerTypes';
+import GameGlobalAPI from '../scenes/gameManager/GameGlobalAPI';
+import SourceAcademyGame from '../SourceAcademyGame';
+import { createButton } from '../utils/ButtonUtils';
+import { sleep } from '../utils/GameUtils';
+import { calcListFormatPos, HexColor } from '../utils/StyleUtils';
+import {
+  questionPrompt,
+  questionTextStyle,
+  QuizConstants,
+  quizOptStyle,
+  quizTextStyle,
+  startPrompt
+} from './GameQuizConstants';
+import GameQuizReactionManager from './GameQuizReactionManager';
+import { Question } from './GameQuizType';
+
+export default class QuizManager {
+  private reactionManager?: GameQuizReactionManager;
+
+  /**
+   * Rendering the quiz section inside a dialogue.
+   *
+   * @param quizId The Id of quiz that users will attempt inside a dialogue.
+   */
+  public async showQuiz(quizId: ItemId) {
+    const quiz = GameGlobalAPI.getInstance().getQuizById(quizId);
+    const numOfQns = quiz.questions.length;
```

Comment:
Nit: I would prefer `numOfQuestions` to be consistent with naming of `question`-related fields.

Reply:
Comment addressed: commit e21f29266c1f39ed6f626af0bbc311e7679178d5